DevCycle.SDK.Server.Common/API/DevCycleProvider.cs

```diff
@@ -0,0 +1,78 @@
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using DevCycle.SDK.Server.Common.Model;
+using Newtonsoft.Json.Linq;
+using OpenFeature;
+using OpenFeature.Constant;
+using OpenFeature.Model;
+
+namespace DevCycle.SDK.Server.Common.API
+{
+    public class DevCycleProvider : FeatureProvider
+    {
+        private DevCycleBaseClient Client { get; }
+
+        public DevCycleProvider(DevCycleBaseClient client)
+        {
+            Client = client;
+        }
+
+        public override Metadata GetMetadata()
+        {
+            return new Metadata(Client.SdkPlatform);
```

Comment:
I'm not sure what we want to set the naming of the provider as - given we have two separate sdk's here that implement the common logic. I have it currently resulting in either `C# Local` or `C# Cloud` as inferred from the client platform.

Reply:
I actually like the idea of differentiating them and I'll pull that in to the Java SDK too, that said who knows how much the MetatData is used outside of debugging?

Reply:
It is used for debugging but also telemetry (for example the standard OTel hooks will add this data to Otel traces if configured to do so).